Shared/Samples/Augment reality to collect data/AugmentRealityToCollectDataView.swift

```diff
@@ -0,0 +1,210 @@
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import SwiftUI
+
+struct AugmentRealityToCollectDataView: View {
+    /// A scene with an imagery basemap.
+    @State private var scene: ArcGIS.Scene = {
+        // Creates an elevation source from Terrain3D REST service.
+        let elevationServiceURL = URL(
+            string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"
+        )!
+        let elevationSource = ArcGISTiledElevationSource(url: elevationServiceURL)
+        let surface = Surface()
+        surface.addElevationSource(elevationSource)
+        surface.backgroundGrid.isVisible = false
+        // Allow camera to go beneath the surface.
+        surface.navigationConstraint = .unconstrained
+        let scene = Scene(basemapStyle: .arcGISImagery)
+        scene.baseSurface = surface
+        scene.baseSurface.opacity = 0
+        return scene
+    }()
+    /// The AR tree survey service feature table.
+    private let featureTable = ServiceFeatureTable(
+        url: URL(string: "https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/AR_Tree_Survey/FeatureServer/0")!
+    )
+    /// The graphics overlay which shows marker symbols.
+    @State private var graphicsOverlay: GraphicsOverlay = {
+        let graphicsOverlay = GraphicsOverlay()
+        let tappedPointSymbol = SimpleMarkerSceneSymbol(
+            style: .diamond,
+            color: .orange,
+            height: 0.5,
+            width: 0.5,
+            depth: 0.5,
+            anchorPosition: .center
+        )
+        graphicsOverlay.renderer = SimpleRenderer(symbol: tappedPointSymbol)
+        graphicsOverlay.sceneProperties.surfacePlacement = .absolute
+        return graphicsOverlay
+    }()
+    /// The status text displayed to the user.
+    @State private var statusText = "Tap to create a feature"
+    /// The selected tree health for the new feature.
+    @State private var treeHealth: TreeHealth?
+    /// A Boolean value indicating whether a feature can be added .
+    @State private var canAddFeature = false
+    /// A Boolean value indicating whether the tree health action sheet is presented.
+    @State private var treeHealthSheetIsPresented = false
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    init() {
+        let featureLayer = FeatureLayer(featureTable: featureTable)
+        featureLayer.sceneProperties.surfacePlacement = .absolute
+        scene.addOperationalLayer(featureLayer)
+    }
+    
+    var body: some View {
+        if #available(iOS 16, *) {
+            NavigationStack {
+                augmentRealityToCollectDataView
+            }
+        } else {
+            NavigationView {
+                augmentRealityToCollectDataView
+            }
+        }
+    }
+    
+    @MainActor @ViewBuilder var augmentRealityToCollectDataView: some View {
```

Comment:
Having to add `MainActor` is highly unusual, but I see that it is because `WorldScaleSceneView`'s initializer is annotated with that (also highly unusual). As such, not anything that can be done about it (at least in the samples app).

Reply:
> I see that it is because WorldScaleSceneView's initializer is annotated with that (also highly unusual)

Can you dig into this and see if there is something that can be done about that?

Reply:
`WorldScaleSceneView` is annotated as `MainActor` because it instantiates a `WorldScaleCalibrationViewModel` which is annotated as `MainActor`. I'm not sure if `WorldScaleCalibrationViewModel` still needs the `MainActor` annotation.

Reply:
ok, please try to remove it and see.

Reply:
The model should be main actor, but it shouldn't be instantiated inside the view's initializer (it shouldn't take any parameters). I'm working on correcting that.

Reply:
The calibration view model works without the `MainActor` annotation, but as Phil suggested the view model could instead be instantiated elsewhere.

Reply:
PR: https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/pull/650